### PR TITLE
added fractional_knapsack in Greedy folder 

### PR DESCRIPTION
**Pull request by LayaMitta**
Fractional knapsack,one of the example of greedy problems.

### CODE REVIEW

File: Algorithm/Greedy/fractional_knapsack.cpp [hunk 1]
```diff
@@ -0,0 +1,101 @@
+#include<bits/stdc++.h>
```
Comment (raksha009) [inline Algorithm/Greedy/fractional_knapsack.cpp:1]: Add a brief description of problem statement

File: Algorithm/Greedy/fractional_knapsack.cpp [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	int ub=0;//current weight in knapsack
+	//An array to store the quantity of ith item that are picked in descending order
+	for(int i=0;i<n;i++)
+		quantity[i]=0.0;
+	for(int i=0;i<n;i++)
+	{
+		if(ub+weight[i]<=knapsack_cap)
+		{
+			//If the value is less than the knapsack capacity,total item is considered
+			quantity[i]=1;
```
Comment (raksha009) [inline Algorithm/Greedy/fractional_knapsack.cpp:17]: Follow 4 space indentation throughout.

File: Algorithm/Greedy/fractional_knapsack.cpp [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+	cin>>n;
+	float weight[n],profit[n],items[n];
+	cout<<"Enter weights of the items"<<endl;
+	for(int i=0;i<n;i++)
+		cin>>weight[i];
+	cout<<"Enter profit for each item"<<endl;
+	for(int i=0;i<n;i++)
+		cin>>profit[i];
+	cout<<"Enter the knapsack capacity"<<endl;
+		cin>>knapsack_cap;
```
Comment (raksha009) [inline Algorithm/Greedy/fractional_knapsack.cpp:52]: indentation

File: Algorithm/Greedy/fractional_knapsack.cpp [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+}
+/* Test case 1:
+   weights[i]= 10 20 30
+   profit[i]= 60 100 120
+   knapsack capacity= 50
+   
+   Output:
+   Total profit 240
+	Items ratio considered:
+	1 1 0.666667
```
Comment (raksha009) [inline Algorithm/Greedy/fractional_knapsack.cpp:90]: Indentation!

**Review by raksha009 — CHANGES_REQUESTED** [review]
Go though the guidelines properly. and link it to the folder.

**Comment by rudrakshi99** [comment]
> Fractional knapsack,one of the example of greedy problems.

link your PR with the issue.

File: Algorithm/Greedy/fractional_knapsack.cpp [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+	
+	TestCase 2:
+	weights[i]=40 10 20 24
+	profit[i]=280 100 120 120
+	knapsack capacity=60
+	
+	Output:
+	Total profit 440
+	Items ratio considered:
+	1 1 0.5 0
```
Comment (rudrakshi99) [inline Algorithm/Greedy/fractional_knapsack.cpp:100]: add space and time complexity.

File: Algorithm/Greedy/fractional_knapsack.cpp (hunk 2 above)
Comment (LayaMitta) [reply Algorithm/Greedy/fractional_knapsack.cpp:22]: Could you please brief me out on what exactly I have to do for 4 space indentation , or can I just dm you?

**Comment by raksha009** [comment]
> Go though the guidelines properly. and link it to the folder.

Make this change and align the test cases and output properly, then we are good to merge.

File: Algorithm/Greedy/fractional_knapsack.cpp (hunk 5 above)
Comment (LayaMitta) [reply Algorithm/Greedy/fractional_knapsack.cpp:105]: Added space and time complexity.

File: Algorithm/Greedy/fractional_knapsack.cpp (hunk 4 above)
Comment (LayaMitta) [reply Algorithm/Greedy/fractional_knapsack.cpp:95]: Done

**Comment by rudrakshi99** [comment]
> > Fractional knapsack,one of the example of greedy problems.
> 
> link your PR with the issue.

@LayaMitta

**Comment by LayaMitta** [comment]
> > > Fractional knapsack,one of the example of greedy problems.
> > 
> > 
> > link your PR with the issue.
> 
> @LayaMitta



> > > Fractional knapsack,one of the example of greedy problems.
> > 
> > 
> > link your PR with the issue.
> 
> @LayaMitta

actually there is no request for fractional knapsack,but I would like to include it in greedy folder.

**Comment by rudrakshi99** [comment]
@LayaMitta Open an issue regarding that and when it will get assigned to you then link this PR with that issue. Otherwise, we will not merge it.

**Comment by raksha009** [comment]
Heyy, extremely sorry. I didn't notice that the issue wasn't linked with the PR and the issue isn't assigned to you. Plz create a PR only when the issue is assigned else it would be unfair for other participants. :)

**Pull request closed without merging**